app/models/manageiq/providers/embedded_ansible/automation_manager/scm_credential.rb

```diff
@@ -66,4 +68,8 @@ def self.params_to_attributes(params)
 
     attrs
   end
+
+  def ensure_newline_for_ssh_key
+    self.auth_key = "#{auth_key}\n" unless auth_key.to_s[-1] == "\n"
```

Comment:
Is the `auth_key` encoding already guaranteed? Just want to make sure `to_s[-1]` doesn't break with an encoding compatibility error.

Reply:
Should we only do this for when the key has the OPENSSH guard kind?

Reply:
Oh I just read the gitter channel, and yeah, if you think that all keys could benefit from a trailing newline, then I'm good with this.

Reply:
Also I can't tell from the diff, but please make this a private method.

Reply:
Oh I just noticed a problem here...if auth_key is nil (for a non-ssh based authentication, then will this change the nil to a blank + \n, which is probably not wanted?

Reply:
> Also I can't tell from the diff, but please make this a private method.

@Fryguy Missed this oringally, but will do.

> Oh I just noticed a problem here...if auth_key is nil

@Fryguy fixed and added specs

> Is the auth_key encoding already guaranteed? Just want to make sure to_s[-1] doesn't break with an encoding compatibility error.

@djberg96 this inherits from `credential.rb`, which inherits from `app/models/authentication.rb`, and that model shows this attributes are covered:

https://github.com/ManageIQ/manageiq/blob/cf52121c1d2f8e881f1ee5f94445e148b878803d/app/models/authentication.rb#L20-L24